### DISCUSSION TITLE
Redirect print statements to a scrollable div?

### DISCUSSION THREAD

**Discussion by gregnordin** [AnswerDotAI/fasthtml#347] [Q&A]
I need to re-purpose some existing python code that gives printed updates during execution. I want to feed these printed lines as they occur to a div that displays the text line-by-line and scrolls once the height of the lines exceeds the height of the div. I've tried a variety of things but without success. Any suggestions?

**Comment by comalice** [comment]
What have you tried? A few ideas occur to me in no particular order:

- override the built-in print method before importing your modules, and target a data structure or similar with your new function
- override the stdout file descriptor to point to your own file descriptor object (that then points to a data structure or pass through function of your choice)

**Comment by gregnordin** [comment]
Thanks for the ideas. Unfortunately, I moved on since I needed to get this done on a short timeline, and it involved running other python code with `subprocess`, the print output of which is what I needed to display. If I could spend more time on it, I would try your idea to override the stdout file descriptor for the subprocess output. In the end, I dropped back to the ever available `tkinter`. I had hoped to use the task to start learning FastHTML, but it looks like that will have to wait for another project.
